rdr_service/dao/consent_dao.py

```diff
@@ -33,29 +38,81 @@ def _get_query_non_test_summaries(cls, session):
+        """
+
+        primary_consent = aliased(ConsentFile)
+        cabor_consent = aliased(ConsentFile)
+        ehr_consent = aliased(ConsentFile)
+        gror_consent = aliased(ConsentFile)
+        primary_update_consent = aliased(ConsentFile)
+        query = (
+            cls._get_query_non_test_summaries(session)
+            .outerjoin(
+                primary_consent, and_(
+                    primary_consent.participant_id == ParticipantSummary.participantId,
+                    primary_consent.type == ConsentType.PRIMARY
                 )
-            )
-        else:
-            query = query.filter(
+            ).outerjoin(
+                cabor_consent, and_(
+                    cabor_consent.participant_id == ParticipantSummary.participantId,
+                    cabor_consent.type == ConsentType.CABOR
+                )
+            ).outerjoin(
+                ehr_consent, and_(
+                    ehr_consent.participant_id == ParticipantSummary.participantId,
+                    ehr_consent.type == ConsentType.EHR
+                )
+            ).outerjoin(
+                gror_consent, and_(
+                    gror_consent.participant_id == ParticipantSummary.participantId,
+                    gror_consent.type == ConsentType.GROR
+                )
+            ).outerjoin(
+                primary_update_consent, and_(
+                    primary_update_consent.participant_id == ParticipantSummary.participantId,
+                    primary_update_consent.type == ConsentType.PRIMARY_UPDATE
+                )
+            ).outerjoin(
+                QuestionnaireResponse,
+                QuestionnaireResponse.participantId == ParticipantSummary.participantId
+            ).outerjoin(
+                QuestionnaireResponseAnswer,
+                QuestionnaireResponseAnswer.questionnaireResponseId == QuestionnaireResponse.questionnaireResponseId
+            ).outerjoin(
+                QuestionnaireQuestion,
+                QuestionnaireQuestion.questionnaireQuestionId == QuestionnaireResponseAnswer.questionId
+            ).outerjoin(
+                Code,
+                and_(
+                    Code.codeId == QuestionnaireQuestion.codeId,
+                    Code.value == PRIMARY_CONSENT_UPDATE_QUESTION_CODE
+                )
+            ).filter(
                 or_(
-                    ParticipantSummary.consentForStudyEnrollmentFirstYesAuthored.between(start_date, end_date),
-                    ParticipantSummary.consentForCABoRAuthored.between(start_date, end_date),
-                    ParticipantSummary.consentForElectronicHealthRecordsAuthored.between(start_date, end_date),
-                    ParticipantSummary.consentForGenomicsRORAuthored.between(start_date, end_date)
+                    primary_consent.id.is_(None),
+                    and_(
+                        ParticipantSummary.consentForCABoR == QuestionnaireStatus.SUBMITTED,
```

Comment:
I assume CABoR is always an implicit yes/SUBMITTED similar to primary consent if a consent response exists, but the EHR and GROR statuses may be something else (SUBMITTED_NO_CONSENT/SUBMITTED_NOT_SURE)?

Reply:
Per offline conversation, apparently don't need to worry about cases where current consentForElectronicHealthRecords  or consentForGenomicsROR  is not / is no longer SUBMITTED.

Reply:
To clarify: the cases where a file is missed when a participant submitted consent and then immediately revoked it may be important and should be brought up in a discussion. But that will require larger changes than to these filters here, and this change doesn't introduce that scenario (it is currently happening).